src/Symfony/Component/Messenger/Transport/AmqpExt/Connection.php

```diff
@@ -151,6 +114,38 @@ public static function fromDsn(string $dsn, array $options = [], AmqpFactory $am
         return new self($amqpOptions, $exchangeOptions, $queuesOptions, $amqpFactory);
     }
 
+    /**
+     * Creates a connection based on the DSN and options.
+     *
+     * Available options:
+     *
+     *   * host: Hostname of the AMQP service
+     *   * port: Port of the AMQP service
+     *   * vhost: Virtual Host to use with the AMQP service
+     *   * user: Username to use to connect the the AMQP service
+     *   * password: Password to use the connect to the AMQP service
+     *   * queues[name]: An array of queues, keyed by the name
+     *     * binding_keys: The binding keys (if any) to bind to this queue
+     *     * flags: Queue flags (Default: AMQP_DURABLE)
+     *     * arguments: Extra arguments
+     *   * exchange:
+     *     * name: Name of the exchange
+     *     * type: Type of exchange (Default: fanout)
+     *     * default_publish_routing_key: Routing key to use when publishing, if none is specified on the message
+     *     * flags: Exchange flags (Default: AMQP_DURABLE)
+     *     * arguments: Extra arguments
+     *   * delay:
+     *     * routing_key_pattern: The pattern of the routing key (Default: "delay_%exchange_name%_%routing_key%_%delay%")
+     *     * queue_name_pattern: Pattern to use to create the queues (Default: "delay_queue_%exchange_name%_%routing_key%_%delay%")
+     *     * exchange_name: Name of the exchange to be used for the retried messages (Default: "delay")
+     *   * auto_setup: Enable or not the auto-setup of queues and exchanges (Default: true)
+     *   * prefetch_count: set channel prefetch count
+     */
+    public static function fromDsn(string $dsn, array $options = [], AmqpFactory $amqpFactory = null): self
```

Comment:
Putting this method before `fromDsnObject` will limit the diff (as we will have added lines in the middle instead of moving the phpdoc describing the options). This will make merging branches easier.